android/iden3coreapi/src/main/java/com/iden3/iden3coreapi/Iden3Identity.kt

```diff
@@ -0,0 +1,155 @@
+     * @return An Iden3Ticket.
+     */
+    @Throws(Exception::class)
+    fun requestClaim(issuerUrl: String, data: String, ticketListener: Iden3TicketListener?) : Iden3Ticket? {
+        if (ticketListener == null) {
+            return Iden3Ticket(identity.requestClaim(issuerUrl, data))
+        } else {
+            identity.requestClaimWithCb(issuerUrl, data) {
+                ticket, exception ->
+                if (exception != null) {
+                    ticketListener.onError(exception)
+                } else if (ticket != null) {
+                    ticketListener.onTicketReceived(Iden3Ticket(ticket))
+                }
+            }
+        }
+        return null
+    }
+
+    /**
+     * Prove a claim from an verifier.
+     *
+     * @param verifierUrl: url of the verifier.
+     * @param credentialId: credential id needed by the verifier for proving a claim.
+     * @param withZKProof: Boolean that indicates if zero-knowledge proof shoud be used or not.
+     * @param proveClaimListener: Listener of proofs received after proving a claim.
+     *
+     * @return Boolean that indicates if the claim has been proven or not.
+     */
+    @Throws(Exception::class)
+    fun proveClaim(verifierUrl: String, credentialId: String, withZKProof: Boolean, proveClaimListener: Iden3ProveClaimListener?) : Boolean?  {
+        if (withZKProof) {
+            if (proveClaimListener == null) {
+                return identity.proveClaim(verifierUrl, credentialId)
+            } else {
+                identity.proveClaimWithCb(verifierUrl, credentialId) {
+                    isProven, exception ->
+                    if (exception != null) {
+                        proveClaimListener.onError(exception)
+                    } else {
+                        proveClaimListener.onClaimProofReceived(isProven)
+                    }
+                }
+            }
+        } else {
+            if (proveClaimListener == null) {
+                return identity.proveClaimZK(verifierUrl, credentialId)
+            } else {
+                identity.proveClaimZKWithCb(verifierUrl, credentialId) {
+                        isProven, exception ->
+                        if (exception != null) {
+                            proveClaimListener.onError(exception)
+                        } else {
+                            proveClaimListener.onClaimProofReceived(isProven)
+                        }
+                }
+            }
+        }
+        return null
+    }
+
+    /**
+     * List of claims of the identity.
+     *
+     * @return ArrayList with the claims of the identity.
+     */
+    @Throws(Exception::class)
+    fun listClaims() : ArrayList<HashMap<*,*>> {
+        val claims = ArrayList<HashMap<*, *>>()
+        try {
+            val cdb = identity.claimDB
+            cdb.iterateClaimsJSON { key, claim ->
+                val map = HashMap<String, Any?>()
+                map["key"] = key
+                map["claim"] = claim
```

Comment:
It's ok for now, but in the future it would be nice to have a class to represent claims (instead of a JSON string). Since there will be deep changes on this in the next iterations, it's a good idea to leave it as it is for now.

Reply:
done, implemented class representing claims